API/DataModel/GenericRepo.cs

```diff
@@ -0,0 +1,135 @@
+﻿using API.DataModel.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace API.DataModel
+{
+    public class GenericRepo<TEntity> : IGenericRepo<TEntity> where TEntity : class
+    {
+        internal DataContext context;
+        internal DbSet<TEntity> dbSet;
+
+        public GenericRepo(DataContext context)
+        {
+            this.context = context;
+            this.dbSet = context.Set<TEntity>();
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> GetAll(
+             Expression<Func<TEntity, bool>> filter = null,
+             Func<IQueryable<TEntity>, IOrderedQueryable<TEntity>> orderBy = null,
+             Func<IQueryable<TEntity>, IIncludableQueryable<TEntity, object>> includes = null
+
+             )
+        {
+            IQueryable<TEntity> query = dbSet;
+
+            if (filter != null)
+            {
+                query = query.Where(filter);
+            }
+
+            if (includes != null)
+            {
+                query = includes(query);
+            }
+
+            if (orderBy != null)
+            {
+                return await orderBy(query).ToListAsync();
+            }
+            else
+            {
+                return await query.ToListAsync();
+            }
+        }
+
+        public virtual async Task<TEntity> Get(Expression<Func<TEntity, bool>> filter = null,
+            Func<IQueryable<TEntity>, IIncludableQueryable<TEntity, object>> includes = null
+
+            )
+        {
+            IQueryable<TEntity> query = dbSet;
+
+            if (filter != null)
+            {
+                query = query.Where(filter);
+            }
+
+            if (includes != null)
+            {
+                query = includes(query);
+            }
+            if (query.Any())
+                return await query.FirstAsync();
```

Comment:
jeżeli get z założenia ma nam zwracać jeden rekord to bym dał SingleAsync zamiast FirstAsync. Wtedy poleci exception jak znajdzie więcej niż 1

Reply:
albo skoro w elsie zwracasz nulla i tak to możesz tu w ogóle zwracać SingleOrDefault async ;) wtedy jak nic nie znajdzie to nulla zwróci

Reply:
SingleOrDefault,  dwie linijki mniej.